Clang build issues

Not immediately sure how to properly fix these:

```
src/thd_engine_adaptive.cpp:428:42: error: result of comparison of constant 4096 with expression of type 'enum adaptive_condition' is always false [-Werror,-Wtautological-constant-out-of-range-compare]
                        } else if (condition_set[j].condition >= 0x1000 && condition_set[j].condition < 0x10000) {
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~
src/thd_engine_adaptive.cpp:428:82: error: result of comparison of constant 65536 with expression of type 'enum adaptive_condition' is always true [-Werror,-Wtautological-constant-out-of-range-compare]
                        } else if (condition_set[j].condition >= 0x1000 && condition_set[j].condition < 0x10000) {
                                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~
src/thd_engine_adaptive.cpp:440:36: error: result of comparison of constant 4 with expression of type 'enum adaptive_comparison' is always true [-Werror,-Wtautological-constant-out-of-range-compare]
                        if (condition_set[j].comparison < ARRAY_SIZE(comp_strs)) {
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~
src/thd_engine_adaptive.cpp:876:29: error: result of comparison of constant 4 with expression of type 'enum adaptive_comparison' is always true [-Werror,-Wtautological-constant-out-of-range-compare]
                        if (condition.comparison < ARRAY_SIZE(comp_strs)) {
                            ~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~
4 errors generated.
```

Are `condition` and `comparison` in `condition` really meant to be enums?

--- a/src/thd_engine_adaptive.cpp
+++ b/src/thd_engine_adaptive.cpp
@@ -846,7 +846,7 @@ int cthd_engine_adaptive::verify_condition(struct condition condition) {
 		return 0;
 
 	cond_name = condition_names[MIN(MAX(0, condition.condition), G_N_ELEMENTS(condition_names) - 1)];
-	thd_log_error("Unsupported condition %d (%s)\n", condition.condition, cond_name);
+	thd_log_error("Unsupported condition %" PRIu64 " (%s)\n", condition.condition, cond_name);
 	return THD_ERROR;
 }
 
@@ -880,11 +880,11 @@ int cthd_engine_adaptive::compare_condition(struct condition condition,
 						cond_name.c_str(), comp_str.c_str(), value);
 			} else {
 				thd_log_debug(
-						"compare condition [%s] comparison [%d] value [%d]\n",
+						"compare condition [%s] comparison [%" PRIu64 "] value [%d]\n",
 						cond_name.c_str(), condition.comparison, value);
 			}
 		} else {
-			thd_log_debug("compare condition %d value %d\n",
+			thd_log_debug("compare condition %" PRIu64 " value %d\n",
 					condition.comparison, value);
 		}
 	}
@@ -1046,7 +1046,7 @@ int cthd_engine_adaptive::evaluate_condition(struct condition condition) {
 	if (condition.condition == Default)
 		return THD_SUCCESS;
 
-	thd_log_debug("evaluate condition.condition %d\n", condition.condition);
+	thd_log_debug("evaluate condition.condition %" PRIu64 "\n", condition.condition);
 
 	if ((condition.condition >= Oem0 && condition.condition <= Oem5)
 			|| (condition.condition >= (adaptive_condition) 0x1000

--- a/src/thd_engine_adaptive.h
+++ b/src/thd_engine_adaptive.h
@@ -119,9 +119,9 @@ struct psv {
 };
 
 struct condition {
-	enum adaptive_condition condition;
+	uint64_t condition;
 	std::string device;
-	enum adaptive_comparison comparison;
+	uint64_t comparison;
 	int argument;
 	enum adaptive_operation operation;
 	enum adaptive_comparison time_comparison;

